### PR TITLE
catch unhandled error when reconnecting with request queue

### PR DESCRIPTION
**Pull request by luu-alex**
## Description

Addresses https://github.com/web3/web3.js/issues/6968
This issue addresses when a WS reconnects, if there are _pendingRequestsQueue in queue. It will send requests. This method can throw an error and will not be caught.
This PR will catch unhandled error

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-utils/src/socket_provider.ts [hunk 1]
```diff
@@ -503,9 +503,15 @@ export abstract class SocketProvider<
 
 	private _sendPendingRequests() {
 		for (const [id, value] of this._pendingRequestsQueue.entries()) {
-			this._sendToSocket(value.payload as Web3APIPayload<API, any>);
-			this._pendingRequestsQueue.delete(id);
-			this._sentRequestsQueue.set(id, value);
+			try {
```
Comment (luu-alex) [inline packages/web3-utils/src/socket_provider.ts:506]: Changes

File: packages/web3-utils/test/unit/socket_provider.test.ts [hunk 2]
```diff
@@ -496,6 +495,34 @@ describe('SocketProvider', () => {
 				expect(deleteSpy).toHaveBeenCalled();
 			});
 		});
+		describe('testing _onConnect() method', () => {
```
Comment (luu-alex) [inline packages/web3-utils/test/unit/socket_provider.test.ts:498]: Adding test

**Comment by luu-alex** [comment]
This PR includes formatting for web3-utils. Noted the actual changes in pr

File: packages/web3-utils/src/socket_provider.ts [hunk 3]
```diff
@@ -503,9 +503,15 @@ export abstract class SocketProvider<
...
+				this._sendToSocket(value.payload as Web3APIPayload<API, any>);
+				this._pendingRequestsQueue.delete(id);
+				this._sentRequestsQueue.set(id, value);
+			} catch (error) {
+				// catches if sendTosocket fails
+				this._pendingRequestsQueue.delete(id);
```
Comment (jdevcs) [inline packages/web3-utils/src/socket_provider.ts:512]: add this in changelog,

**Review by jdevcs — APPROVED** [review]

File: packages/web3-utils/src/socket_provider.ts (hunk 3 above)
Comment (luu-alex) [reply packages/web3-utils/src/socket_provider.ts:512]: Updated, thanks!

**Pull request merged**